packages/docs/pages/integrating-with-namada/sdk/shielded-transfers.mdx

```diff
@@ -56,4 +61,33 @@ async fn construct_transfer(sdk: impl Namada) {
       Err(e) => println!("Tx error: {:?}", e)
     }
 }
-```
\ No newline at end of file
+```
+
+### Frontend sustainability fee
+
+For transactions that move assets into or out of the shielded pool, the namada
+SDK provides the possibility to charge an optional fee to support the operations
+of the frontend provider. The specific transactions are:
+
+- `TxShieldingTransfer`
+- `TxUnshieldingTransfer`
+- `TxOsmosisSwap`
+- `TxIbcTransfer`
+- `GenIbcShieldingTransfer`
+
+<Callout type="warning">
+  For fully shielded osmosis swaps (from shieleded address to shielded address),
+  the operator should refrain from charging a fee.
+</Callout>
+
+These structures provide an optional field in one of two forms:
+
+```rust
+  pub frontend_sus_fee: Option<(C::TransferTarget, Dec)>,
+  pub frontend_sus_fee: Option<(C::PaymentAddress, Dec)>,
+```
+
+where the first element of the tuple is the receiver of the fee (an address
+controlled by the operator) and the second one is the fee expressed as a
+percentage of the shielded/unshielded amount. This percentage will be charged on
+top of the amount specified by the user.
```

Comment:
Can you write some more details about what the operator can expect to happen each time a user transacts with the MASP? For example, will an extra tx be generated and asked to be signed that pays fees?